{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "refseq_metadata = pd.read_csv(\"assembly_summary.txt\",sep='\\t',skiprows=[0],dtype='str')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#assembly_accession', 'bioproject', 'biosample', 'wgs_master',\n",
       "       'refseq_category', 'taxid', 'species_taxid', 'organism_name',\n",
       "       'infraspecific_name', 'isolate', 'version_status', 'assembly_level',\n",
       "       'release_type', 'genome_rep', 'seq_rel_date', 'asm_name',\n",
       "       'asm_submitter', 'gbrs_paired_asm', 'paired_asm_comp', 'ftp_path',\n",
       "       'excluded_from_refseq', 'relation_to_type_material',\n",
       "       'asm_not_live_date', 'assembly_type', 'group', 'genome_size',\n",
       "       'genome_size_ungapped', 'gc_percent', 'replicon_count',\n",
       "       'scaffold_count', 'contig_count', 'annotation_provider',\n",
       "       'annotation_name', 'annotation_date', 'total_gene_count',\n",
       "       'protein_coding_gene_count', 'non_coding_gene_count', 'pubmed_id'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq_metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "patric_summary = pd.read_csv(\"genome_summary\",sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>genome_length</th>\n",
       "      <th>gc_content</th>\n",
       "      <th>contig_l50</th>\n",
       "      <th>contig_n50</th>\n",
       "      <th>chromosomes</th>\n",
       "      <th>plasmids</th>\n",
       "      <th>contigs</th>\n",
       "      <th>patric_cds</th>\n",
       "      <th>refseq_cds</th>\n",
       "      <th>trna</th>\n",
       "      <th>rrnacoarse_consistency</th>\n",
       "      <th>fine_consistency</th>\n",
       "      <th>checkm_completeness</th>\n",
       "      <th>checkm_contamination</th>\n",
       "      <th>genome_qualitydate_created</th>\n",
       "      <th>date_modified</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>469009.4</td>\n",
       "      <td>\"'Brassica napus' phytoplasma strain TW1\"</td>\n",
       "      <td>469009</td>\n",
       "      <td>WGS</td>\n",
       "      <td>743598</td>\n",
       "      <td>27.23313</td>\n",
       "      <td>1.0</td>\n",
       "      <td>373899.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>1073.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98.2</td>\n",
       "      <td>100.0</td>\n",
       "      <td>10.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2018-07-15T20:38:31.410Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>3236984</td>\n",
       "      <td>70.20000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3236984.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>3197.0</td>\n",
       "      <td>2944.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98.1</td>\n",
       "      <td>96.9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-01-17T16:02:31.274Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1123738.3</td>\n",
       "      <td>\"'Echinacea purpurea' witches'-broom phytoplas...</td>\n",
       "      <td>1123738</td>\n",
       "      <td>WGS</td>\n",
       "      <td>545427</td>\n",
       "      <td>23.89000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>46697.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28</td>\n",
       "      <td>535.0</td>\n",
       "      <td>433.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>94.6</td>\n",
       "      <td>84.0</td>\n",
       "      <td>6.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-01-17T17:13:35.332Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>5486145</td>\n",
       "      <td>38.30000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5354700.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3</td>\n",
       "      <td>7014.0</td>\n",
       "      <td>3651.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.2</td>\n",
       "      <td>99.3</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1856298.3</td>\n",
       "      <td>\"'Osedax' symbiont bacterium Rs2_46_30_T18 str...</td>\n",
       "      <td>1856298</td>\n",
       "      <td>WGS</td>\n",
       "      <td>4021833</td>\n",
       "      <td>45.76000</td>\n",
       "      <td>63.0</td>\n",
       "      <td>19390.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>365</td>\n",
       "      <td>4118.0</td>\n",
       "      <td>3690.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.3</td>\n",
       "      <td>98.7</td>\n",
       "      <td>3.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2017-07-22T04:04:17.224Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059041</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>884635</td>\n",
       "      <td>46.70000</td>\n",
       "      <td>11.0</td>\n",
       "      <td>19981.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>122</td>\n",
       "      <td>946.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>84.8</td>\n",
       "      <td>32.2</td>\n",
       "      <td>3.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059042</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>987314</td>\n",
       "      <td>46.22000</td>\n",
       "      <td>22.0</td>\n",
       "      <td>14661.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>128</td>\n",
       "      <td>1016.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>87.7</td>\n",
       "      <td>24.3</td>\n",
       "      <td>17.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059043</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>451985</td>\n",
       "      <td>46.18000</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5881.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>101</td>\n",
       "      <td>470.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>93.4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059044</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>944583</td>\n",
       "      <td>50.33000</td>\n",
       "      <td>19.0</td>\n",
       "      <td>15090.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>132</td>\n",
       "      <td>1076.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>88.1</td>\n",
       "      <td>44.6</td>\n",
       "      <td>4.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059045</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>979689</td>\n",
       "      <td>50.50000</td>\n",
       "      <td>11.0</td>\n",
       "      <td>23039.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>97</td>\n",
       "      <td>1086.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>19.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>85.7</td>\n",
       "      <td>42.4</td>\n",
       "      <td>6.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2015-03-16T03:17:09.594Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1059046 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         genome_id                                        genome_name  \\\n",
       "0         469009.4          \"'Brassica napus' phytoplasma strain TW1\"   \n",
       "1        1309411.5     \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "2        1123738.3  \"'Echinacea purpurea' witches'-broom phytoplas...   \n",
       "3         551115.6                            \"'Nostoc azollae' 0708\"   \n",
       "4        1856298.3  \"'Osedax' symbiont bacterium Rs2_46_30_T18 str...   \n",
       "...            ...                                                ...   \n",
       "1059041  1131286.3               zeta proteobacterium SCGC AB-137-J06   \n",
       "1059042  1131287.3               zeta proteobacterium SCGC AB-602-C20   \n",
       "1059043  1131288.3               zeta proteobacterium SCGC AB-602-E04   \n",
       "1059044  1131289.3               zeta proteobacterium SCGC AB-604-B04   \n",
       "1059045  1131290.3               zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "         taxon_id genome_status  genome_length  gc_content  contig_l50  \\\n",
       "0          469009           WGS         743598    27.23313         1.0   \n",
       "1         1309411      Complete        3236984    70.20000         1.0   \n",
       "2         1123738           WGS         545427    23.89000         4.0   \n",
       "3          551115      Complete        5486145    38.30000         1.0   \n",
       "4         1856298           WGS        4021833    45.76000        63.0   \n",
       "...           ...           ...            ...         ...         ...   \n",
       "1059041   1131286           WGS         884635    46.70000        11.0   \n",
       "1059042   1131287           WGS         987314    46.22000        22.0   \n",
       "1059043   1131288           WGS         451985    46.18000        18.0   \n",
       "1059044   1131289           WGS         944583    50.33000        19.0   \n",
       "1059045   1131290           WGS         979689    50.50000        11.0   \n",
       "\n",
       "         contig_n50  chromosomes  plasmids  contigs  patric_cds  refseq_cds  \\\n",
       "0          373899.0          NaN       NaN        5      1073.0       705.0   \n",
       "1         3236984.0          1.0       NaN        1      3197.0      2944.0   \n",
       "2           46697.0          NaN       NaN       28       535.0       433.0   \n",
       "3         5354700.0          1.0       2.0        3      7014.0      3651.0   \n",
       "4           19390.0          NaN       NaN      365      4118.0      3690.0   \n",
       "...             ...          ...       ...      ...         ...         ...   \n",
       "1059041     19981.0          0.0       0.0      122       946.0         0.0   \n",
       "1059042     14661.0          0.0       0.0      128      1016.0         0.0   \n",
       "1059043      5881.0          0.0       0.0      101       470.0         0.0   \n",
       "1059044     15090.0          0.0       0.0      132      1076.0         0.0   \n",
       "1059045     23039.0          0.0       0.0       97      1086.0         0.0   \n",
       "\n",
       "         trna  rrnacoarse_consistency  fine_consistency  checkm_completeness  \\\n",
       "0        27.0                     NaN              98.2                100.0   \n",
       "1        50.0                     NaN              98.1                 96.9   \n",
       "2        26.0                     NaN              94.6                 84.0   \n",
       "3         0.0                     NaN              99.2                 99.3   \n",
       "4        59.0                     NaN              88.3                 98.7   \n",
       "...       ...                     ...               ...                  ...   \n",
       "1059041   8.0                     NaN              84.8                 32.2   \n",
       "1059042  12.0                     NaN              87.7                 24.3   \n",
       "1059043   3.0                     NaN              93.4                  8.0   \n",
       "1059044  11.0                     NaN              88.1                 44.6   \n",
       "1059045  19.0                     NaN              85.7                 42.4   \n",
       "\n",
       "         checkm_contamination  genome_qualitydate_created  \\\n",
       "0                        10.3                         NaN   \n",
       "1                         1.0                         NaN   \n",
       "2                         6.4                         NaN   \n",
       "3                         0.7                         NaN   \n",
       "4                         3.1                         NaN   \n",
       "...                       ...                         ...   \n",
       "1059041                   3.1                         NaN   \n",
       "1059042                  17.0                         NaN   \n",
       "1059043                  11.0                         NaN   \n",
       "1059044                   4.3                         NaN   \n",
       "1059045                   6.6                         NaN   \n",
       "\n",
       "                    date_modified  \n",
       "0        2018-07-15T20:38:31.410Z  \n",
       "1        2016-01-17T16:02:31.274Z  \n",
       "2        2016-01-17T17:13:35.332Z  \n",
       "3        2015-03-16T03:17:09.594Z  \n",
       "4        2017-07-22T04:04:17.224Z  \n",
       "...                           ...  \n",
       "1059041  2015-03-16T03:17:09.594Z  \n",
       "1059042  2015-03-16T03:17:09.594Z  \n",
       "1059043  2015-03-16T03:17:09.594Z  \n",
       "1059044  2015-03-16T03:17:09.594Z  \n",
       "1059045  2015-03-16T03:17:09.594Z  \n",
       "\n",
       "[1059046 rows x 20 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "patric_metadata = pd.read_csv(\"genome_metadata\",sep='\\t',dtype='str')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>469009.4</td>\n",
       "      <td>\"'Brassica napus' phytoplasma strain TW1\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>469009</td>\n",
       "      <td>WGS</td>\n",
       "      <td>TW1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of a strain of bacteria that c...</td>\n",
       "      <td>sample_type:metagenomic assembly;collected_by:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>N5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing of a Gamma-Radiation-Resista...</td>\n",
       "      <td>sample_type:bacterial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1123738.3</td>\n",
       "      <td>\"'Echinacea purpurea' witches'-broom phytoplas...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1123738</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NCHU2014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>'Echinacea purpurea' witches'-broom phytoplasm...</td>\n",
       "      <td>lab_host:Catharanthus roseus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>'Nostoc azollae' 0708</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nostoc azollae 0708. Nostoc azollae 0708, also...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1856298.3</td>\n",
       "      <td>\"'Osedax' symbiont bacterium Rs2_46_30_T18 str...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1856298</td>\n",
       "      <td>WGS</td>\n",
       "      <td>Rs2_46_30_T18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>In this study, we simulate the Deepwater Horiz...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059041</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-137-J06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059042</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-C20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059043</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-E04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059044</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-B04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059045</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-O16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1059046 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         genome_id                                        genome_name  \\\n",
       "0         469009.4          \"'Brassica napus' phytoplasma strain TW1\"   \n",
       "1        1309411.5     \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "2        1123738.3  \"'Echinacea purpurea' witches'-broom phytoplas...   \n",
       "3         551115.6                            \"'Nostoc azollae' 0708\"   \n",
       "4        1856298.3  \"'Osedax' symbiont bacterium Rs2_46_30_T18 str...   \n",
       "...            ...                                                ...   \n",
       "1059041  1131286.3               zeta proteobacterium SCGC AB-137-J06   \n",
       "1059042  1131287.3               zeta proteobacterium SCGC AB-602-C20   \n",
       "1059043  1131288.3               zeta proteobacterium SCGC AB-602-E04   \n",
       "1059044  1131289.3               zeta proteobacterium SCGC AB-604-B04   \n",
       "1059045  1131290.3               zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "                                organism_name taxon_id genome_status  \\\n",
       "0                                         NaN   469009           WGS   \n",
       "1                                         NaN  1309411      Complete   \n",
       "2                                         NaN  1123738           WGS   \n",
       "3                       'Nostoc azollae' 0708   551115      Complete   \n",
       "4                                         NaN  1856298           WGS   \n",
       "...                                       ...      ...           ...   \n",
       "1059041  zeta proteobacterium SCGC AB-137-J06  1131286           WGS   \n",
       "1059042  zeta proteobacterium SCGC AB-602-C20  1131287           WGS   \n",
       "1059043  zeta proteobacterium SCGC AB-602-E04  1131288           WGS   \n",
       "1059044  zeta proteobacterium SCGC AB-604-B04  1131289           WGS   \n",
       "1059045  zeta proteobacterium SCGC AB-604-O16  1131290           WGS   \n",
       "\n",
       "                  strain serovar biovar pathovar mlst  ... motility  \\\n",
       "0                    TW1     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1                     N5     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "2               NCHU2014     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "3                    708     NaN    NaN      NaN  NaN  ...      Yes   \n",
       "4          Rs2_46_30_T18     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "...                  ...     ...    ...      ...  ...  ...      ...   \n",
       "1059041  SCGC AB-137-J06     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059042  SCGC AB-602-C20     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059043  SCGC AB-602-E04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059044  SCGC AB-604-B04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059045  SCGC AB-604-O16     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "\n",
       "        sporulation temperature_range optimal_temperature salinity  \\\n",
       "0               NaN               NaN                 NaN      NaN   \n",
       "1               NaN               NaN                 NaN      NaN   \n",
       "2               NaN               NaN                   C      NaN   \n",
       "3               NaN        Mesophilic                   -      NaN   \n",
       "4               NaN               NaN                 NaN      NaN   \n",
       "...             ...               ...                 ...      ...   \n",
       "1059041         NaN               NaN                 NaN      NaN   \n",
       "1059042         NaN               NaN                 NaN      NaN   \n",
       "1059043         NaN               NaN                 NaN      NaN   \n",
       "1059044         NaN               NaN                 NaN      NaN   \n",
       "1059045         NaN               NaN                 NaN      NaN   \n",
       "\n",
       "        oxygen_requirement   habitat disease  \\\n",
       "0                      NaN       NaN     NaN   \n",
       "1                      NaN       NaN     NaN   \n",
       "2                      NaN       NaN     NaN   \n",
       "3                  Aerobic  Multiple     NaN   \n",
       "4                      NaN       NaN     NaN   \n",
       "...                    ...       ...     ...   \n",
       "1059041                NaN       NaN     NaN   \n",
       "1059042                NaN       NaN     NaN   \n",
       "1059043                NaN       NaN     NaN   \n",
       "1059044                NaN       NaN     NaN   \n",
       "1059045                NaN       NaN     NaN   \n",
       "\n",
       "                                                  comments  \\\n",
       "0        Genome sequence of a strain of bacteria that c...   \n",
       "1        Genome sequencing of a Gamma-Radiation-Resista...   \n",
       "2        'Echinacea purpurea' witches'-broom phytoplasm...   \n",
       "3        Nostoc azollae 0708. Nostoc azollae 0708, also...   \n",
       "4        In this study, we simulate the Deepwater Horiz...   \n",
       "...                                                    ...   \n",
       "1059041  Single cell genome sequencing of biomineralizi...   \n",
       "1059042  Single cell genome sequencing of biomineralizi...   \n",
       "1059043  Single cell genome sequencing of biomineralizi...   \n",
       "1059044  Single cell genome sequencing of biomineralizi...   \n",
       "1059045  Single cell genome sequencing of biomineralizi...   \n",
       "\n",
       "                                       additional_metadata  \n",
       "0        sample_type:metagenomic assembly;collected_by:...  \n",
       "1                                    sample_type:bacterial  \n",
       "2                             lab_host:Catharanthus roseus  \n",
       "3                                                      NaN  \n",
       "4                         sample_type:metagenomic assembly  \n",
       "...                                                    ...  \n",
       "1059041                                                NaN  \n",
       "1059042                                                NaN  \n",
       "1059043                                                NaN  \n",
       "1059044                                                NaN  \n",
       "1059045                                                NaN  \n",
       "\n",
       "[1059046 rows x 66 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                                    NaN\n",
       "1                          NZ_CP011389.1\n",
       "2                                    NaN\n",
       "3          NC_014248,NC_014249,NC_014250\n",
       "4                                    NaN\n",
       "                       ...              \n",
       "1059041                                -\n",
       "1059042                                -\n",
       "1059043                                -\n",
       "1059044                                -\n",
       "1059045                                -\n",
       "Name: refseq_accessions, Length: 1059046, dtype: object"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata['refseq_accessions']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1309411.5</td>\n",
       "      <td>\"'Deinococcus soli' Cha et al. 2014 strain N5\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1309411</td>\n",
       "      <td>Complete</td>\n",
       "      <td>N5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing of a Gamma-Radiation-Resista...</td>\n",
       "      <td>sample_type:bacterial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>551115.6</td>\n",
       "      <td>\"'Nostoc azollae' 0708\"</td>\n",
       "      <td>'Nostoc azollae' 0708</td>\n",
       "      <td>551115</td>\n",
       "      <td>Complete</td>\n",
       "      <td>708</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Multiple</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nostoc azollae 0708. Nostoc azollae 0708, also...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>483218.5</td>\n",
       "      <td>\"[Bacteroides] pectinophilus ATCC 43243\"</td>\n",
       "      <td>Bacteroides pectinophilus ATCC 43243</td>\n",
       "      <td>483218</td>\n",
       "      <td>WGS</td>\n",
       "      <td>ATCC 43243</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>Mesophilic</td>\n",
       "      <td>35-40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anaerobic</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Bacteroides pectinophilus  ATCC 43243. Bactero...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1232385.4</td>\n",
       "      <td>\"[Brevibacterium] flavum ZL-1\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1232385</td>\n",
       "      <td>Complete</td>\n",
       "      <td>ZL-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Produces lysine, L-lysine.</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>92706.3</td>\n",
       "      <td>\"[Brevibacterium] flavum strain ATCC 15168\"</td>\n",
       "      <td>NaN</td>\n",
       "      <td>92706</td>\n",
       "      <td>Complete</td>\n",
       "      <td>ATCC 15168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>metabolic engineering for valine production</td>\n",
       "      <td>sample_type:Single cell</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059041</th>\n",
       "      <td>1131286.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>zeta proteobacterium SCGC AB-137-J06</td>\n",
       "      <td>1131286</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-137-J06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059042</th>\n",
       "      <td>1131287.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-C20</td>\n",
       "      <td>1131287</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-C20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059043</th>\n",
       "      <td>1131288.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-602-E04</td>\n",
       "      <td>1131288</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-602-E04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059044</th>\n",
       "      <td>1131289.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-B04</td>\n",
       "      <td>1131289</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-B04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059045</th>\n",
       "      <td>1131290.3</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>zeta proteobacterium SCGC AB-604-O16</td>\n",
       "      <td>1131290</td>\n",
       "      <td>WGS</td>\n",
       "      <td>SCGC AB-604-O16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Single cell genome sequencing of biomineralizi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31666 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         genome_id                                     genome_name  \\\n",
       "1        1309411.5  \"'Deinococcus soli' Cha et al. 2014 strain N5\"   \n",
       "3         551115.6                         \"'Nostoc azollae' 0708\"   \n",
       "34        483218.5        \"[Bacteroides] pectinophilus ATCC 43243\"   \n",
       "35       1232385.4                  \"[Brevibacterium] flavum ZL-1\"   \n",
       "36         92706.3     \"[Brevibacterium] flavum strain ATCC 15168\"   \n",
       "...            ...                                             ...   \n",
       "1059041  1131286.3            zeta proteobacterium SCGC AB-137-J06   \n",
       "1059042  1131287.3            zeta proteobacterium SCGC AB-602-C20   \n",
       "1059043  1131288.3            zeta proteobacterium SCGC AB-602-E04   \n",
       "1059044  1131289.3            zeta proteobacterium SCGC AB-604-B04   \n",
       "1059045  1131290.3            zeta proteobacterium SCGC AB-604-O16   \n",
       "\n",
       "                                organism_name taxon_id genome_status  \\\n",
       "1                                         NaN  1309411      Complete   \n",
       "3                       'Nostoc azollae' 0708   551115      Complete   \n",
       "34       Bacteroides pectinophilus ATCC 43243   483218           WGS   \n",
       "35                                        NaN  1232385      Complete   \n",
       "36                                        NaN    92706      Complete   \n",
       "...                                       ...      ...           ...   \n",
       "1059041  zeta proteobacterium SCGC AB-137-J06  1131286           WGS   \n",
       "1059042  zeta proteobacterium SCGC AB-602-C20  1131287           WGS   \n",
       "1059043  zeta proteobacterium SCGC AB-602-E04  1131288           WGS   \n",
       "1059044  zeta proteobacterium SCGC AB-604-B04  1131289           WGS   \n",
       "1059045  zeta proteobacterium SCGC AB-604-O16  1131290           WGS   \n",
       "\n",
       "                  strain serovar biovar pathovar mlst  ... motility  \\\n",
       "1                     N5     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "3                    708     NaN    NaN      NaN  NaN  ...      Yes   \n",
       "34            ATCC 43243     NaN    NaN      NaN  NaN  ...       No   \n",
       "35                  ZL-1     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "36            ATCC 15168     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "...                  ...     ...    ...      ...  ...  ...      ...   \n",
       "1059041  SCGC AB-137-J06     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059042  SCGC AB-602-C20     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059043  SCGC AB-602-E04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059044  SCGC AB-604-B04     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "1059045  SCGC AB-604-O16     NaN    NaN      NaN  NaN  ...      NaN   \n",
       "\n",
       "        sporulation temperature_range optimal_temperature salinity  \\\n",
       "1               NaN               NaN                 NaN      NaN   \n",
       "3               NaN        Mesophilic                   -      NaN   \n",
       "34               No        Mesophilic               35-40      NaN   \n",
       "35              NaN               NaN                 NaN      NaN   \n",
       "36              NaN               NaN                 NaN      NaN   \n",
       "...             ...               ...                 ...      ...   \n",
       "1059041         NaN               NaN                 NaN      NaN   \n",
       "1059042         NaN               NaN                 NaN      NaN   \n",
       "1059043         NaN               NaN                 NaN      NaN   \n",
       "1059044         NaN               NaN                 NaN      NaN   \n",
       "1059045         NaN               NaN                 NaN      NaN   \n",
       "\n",
       "        oxygen_requirement          habitat disease  \\\n",
       "1                      NaN              NaN     NaN   \n",
       "3                  Aerobic         Multiple     NaN   \n",
       "34               Anaerobic  Host-associated     NaN   \n",
       "35                     NaN              NaN     NaN   \n",
       "36                     NaN              NaN     NaN   \n",
       "...                    ...              ...     ...   \n",
       "1059041                NaN              NaN     NaN   \n",
       "1059042                NaN              NaN     NaN   \n",
       "1059043                NaN              NaN     NaN   \n",
       "1059044                NaN              NaN     NaN   \n",
       "1059045                NaN              NaN     NaN   \n",
       "\n",
       "                                                  comments  \\\n",
       "1        Genome sequencing of a Gamma-Radiation-Resista...   \n",
       "3        Nostoc azollae 0708. Nostoc azollae 0708, also...   \n",
       "34       Bacteroides pectinophilus  ATCC 43243. Bactero...   \n",
       "35                              Produces lysine, L-lysine.   \n",
       "36             metabolic engineering for valine production   \n",
       "...                                                    ...   \n",
       "1059041  Single cell genome sequencing of biomineralizi...   \n",
       "1059042  Single cell genome sequencing of biomineralizi...   \n",
       "1059043  Single cell genome sequencing of biomineralizi...   \n",
       "1059044  Single cell genome sequencing of biomineralizi...   \n",
       "1059045  Single cell genome sequencing of biomineralizi...   \n",
       "\n",
       "             additional_metadata  \n",
       "1          sample_type:bacterial  \n",
       "3                            NaN  \n",
       "34                           NaN  \n",
       "35                           NaN  \n",
       "36       sample_type:Single cell  \n",
       "...                          ...  \n",
       "1059041                      NaN  \n",
       "1059042                      NaN  \n",
       "1059043                      NaN  \n",
       "1059044                      NaN  \n",
       "1059045                      NaN  \n",
       "\n",
       "[31666 rows x 66 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata_filtered = patric_metadata[patric_metadata['refseq_accessions'].notna()]\n",
    "patric_metadata_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          GCA_003181115.1\n",
       "1          GCF_001007995.1\n",
       "2          GCF_001307505.1\n",
       "3          GCA_000196515.1\n",
       "4          GCA_002163025.1\n",
       "                ...       \n",
       "1059041    GCA_000379245.1\n",
       "1059042    GCA_000379345.1\n",
       "1059043    GCA_000379265.1\n",
       "1059044    GCA_000379205.1\n",
       "1059045    GCA_000372125.1\n",
       "Name: assembly_accession, Length: 828228, dtype: object"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata_filtered = patric_metadata[patric_metadata['assembly_accession'].notna()]\n",
    "patric_metadata_filtered['assembly_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0         GCF_036600855.1\n",
       "1         GCF_036600875.1\n",
       "2         GCF_036600895.1\n",
       "3         GCF_036600915.1\n",
       "4         GCF_900128725.1\n",
       "               ...       \n",
       "404913    GCF_046824975.1\n",
       "404914    GCF_028826155.1\n",
       "404915    GCF_046825355.1\n",
       "404916    GCF_037013405.1\n",
       "404917    GCF_042466685.1\n",
       "Name: #assembly_accession, Length: 404918, dtype: object"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq_metadata_filtered = refseq_metadata[refseq_metadata['#assembly_accession'].notna()]\n",
    "refseq_metadata_filtered['#assembly_accession']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name_x</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_900128725.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA4556317</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>9</td>\n",
       "      <td>9</td>\n",
       "      <td>Buchnera aphidicola</td>\n",
       "      <td>strain=BCifornacula</td>\n",
       "      <td>2912</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Within the Lachninae subfamily of aphids, a pu...</td>\n",
       "      <td>collected_by:Armelle Coeur D'Acier\\; Emmanuell...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_001589285.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02639847</td>\n",
       "      <td>JEMA00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>56</td>\n",
       "      <td>56</td>\n",
       "      <td>Sorangium cellulosum</td>\n",
       "      <td>strain=So0008-312</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sorangium cellulosum comparative genomic analysis</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_001442515.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04158597</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>69</td>\n",
       "      <td>69</td>\n",
       "      <td>Lysobacter enzymogenes</td>\n",
       "      <td>strain=C3</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The genus Lysobacter consists of several plant...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_001598195.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04544264</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>140</td>\n",
       "      <td>140</td>\n",
       "      <td>Borrelia hermsii</td>\n",
       "      <td>strain=DAH-2E7</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Borrelia hermsii is the causative agent of tic...</td>\n",
       "      <td>collected_by:Tom Schwan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_001012915.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN03283812</td>\n",
       "      <td>JXNB00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>159</td>\n",
       "      <td>159</td>\n",
       "      <td>Brachyspira hyodysenteriae</td>\n",
       "      <td>strain=FMV89.3323</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>confirms that the species is relatively conser...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42778</th>\n",
       "      <td>GCF_001690235.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05179583</td>\n",
       "      <td>LZGO00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>3120262</td>\n",
       "      <td>3120262</td>\n",
       "      <td>Gilliamella sp. wkB292</td>\n",
       "      <td>strain=wkB292</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of Gilliamella apicola strains...</td>\n",
       "      <td>sample_type:cell culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42779</th>\n",
       "      <td>GCF_001693435.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN05179591</td>\n",
       "      <td>LZGG00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>3120264</td>\n",
       "      <td>3120264</td>\n",
       "      <td>Gilliamella sp. wkB7</td>\n",
       "      <td>strain=wkB7</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequence of Gilliamella apicola strains...</td>\n",
       "      <td>sample_type:cell culture</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42780</th>\n",
       "      <td>GCF_000708695.2</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02644483</td>\n",
       "      <td>JENC00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>3344738</td>\n",
       "      <td>219574</td>\n",
       "      <td>Pseudomonas [fluorescens] ATCC 17400</td>\n",
       "      <td>strain=ATCC 17400</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26-28C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Aerobic</td>\n",
       "      <td>Terrestrial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Whole genome sequence of P. fluorecens ATCC 17400</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42781</th>\n",
       "      <td>GCF_001431665.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04207868</td>\n",
       "      <td>LLXV00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>3384461</td>\n",
       "      <td>3384461</td>\n",
       "      <td>Stenotrophomonas beteli</td>\n",
       "      <td>strain=LMG 978</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>To get insight into origin and evolution of St...</td>\n",
       "      <td>sample_type:whole organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42782</th>\n",
       "      <td>GCF_001644665.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02954477</td>\n",
       "      <td>na</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>3389972</td>\n",
       "      <td>3389972</td>\n",
       "      <td>Fervidobacterium ngatamarikiense</td>\n",
       "      <td>strain=DYC</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A thermophilic strict anaerobe capable of cell...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>42783 rows × 104 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #assembly_accession   bioproject     biosample      wgs_master  \\\n",
       "0         GCF_900128725.1  PRJNA224116  SAMEA4556317              na   \n",
       "1         GCF_001589285.1  PRJNA224116  SAMN02639847  JEMA00000000.1   \n",
       "2         GCF_001442515.1  PRJNA224116  SAMN04158597              na   \n",
       "3         GCF_001598195.1  PRJNA224116  SAMN04544264              na   \n",
       "4         GCF_001012915.1  PRJNA224116  SAMN03283812  JXNB00000000.1   \n",
       "...                   ...          ...           ...             ...   \n",
       "42778     GCF_001690235.1  PRJNA224116  SAMN05179583  LZGO00000000.1   \n",
       "42779     GCF_001693435.1  PRJNA224116  SAMN05179591  LZGG00000000.1   \n",
       "42780     GCF_000708695.2  PRJNA224116  SAMN02644483  JENC00000000.1   \n",
       "42781     GCF_001431665.1  PRJNA224116  SAMN04207868  LLXV00000000.1   \n",
       "42782     GCF_001644665.1  PRJNA224116  SAMN02954477              na   \n",
       "\n",
       "        refseq_category    taxid species_taxid  \\\n",
       "0                    na        9             9   \n",
       "1                    na       56            56   \n",
       "2                    na       69            69   \n",
       "3                    na      140           140   \n",
       "4                    na      159           159   \n",
       "...                 ...      ...           ...   \n",
       "42778                na  3120262       3120262   \n",
       "42779                na  3120264       3120264   \n",
       "42780                na  3344738        219574   \n",
       "42781  reference genome  3384461       3384461   \n",
       "42782  reference genome  3389972       3389972   \n",
       "\n",
       "                            organism_name_x   infraspecific_name isolate  ...  \\\n",
       "0                       Buchnera aphidicola  strain=BCifornacula    2912  ...   \n",
       "1                      Sorangium cellulosum    strain=So0008-312      na  ...   \n",
       "2                    Lysobacter enzymogenes            strain=C3      na  ...   \n",
       "3                          Borrelia hermsii       strain=DAH-2E7      na  ...   \n",
       "4                Brachyspira hyodysenteriae    strain=FMV89.3323      na  ...   \n",
       "...                                     ...                  ...     ...  ...   \n",
       "42778                Gilliamella sp. wkB292        strain=wkB292      na  ...   \n",
       "42779                  Gilliamella sp. wkB7          strain=wkB7      na  ...   \n",
       "42780  Pseudomonas [fluorescens] ATCC 17400    strain=ATCC 17400      na  ...   \n",
       "42781               Stenotrophomonas beteli       strain=LMG 978      na  ...   \n",
       "42782      Fervidobacterium ngatamarikiense           strain=DYC      na  ...   \n",
       "\n",
       "      motility sporulation temperature_range optimal_temperature salinity  \\\n",
       "0          NaN         NaN               NaN                 NaN      NaN   \n",
       "1          NaN         NaN               NaN                 NaN      NaN   \n",
       "2          NaN         NaN               NaN                 NaN      NaN   \n",
       "3          NaN         NaN               NaN                 NaN      NaN   \n",
       "4          NaN         NaN               NaN                 NaN      NaN   \n",
       "...        ...         ...               ...                 ...      ...   \n",
       "42778      NaN         NaN               NaN                 NaN      NaN   \n",
       "42779      NaN         NaN               NaN                 NaN      NaN   \n",
       "42780      Yes         NaN               NaN              26-28C      NaN   \n",
       "42781      NaN         NaN               NaN                   C      NaN   \n",
       "42782      NaN         NaN               NaN                 NaN      NaN   \n",
       "\n",
       "      oxygen_requirement      habitat disease  \\\n",
       "0                    NaN          NaN     NaN   \n",
       "1                    NaN          NaN     NaN   \n",
       "2                    NaN          NaN     NaN   \n",
       "3                    NaN          NaN     NaN   \n",
       "4                    NaN          NaN     NaN   \n",
       "...                  ...          ...     ...   \n",
       "42778                NaN          NaN     NaN   \n",
       "42779                NaN          NaN     NaN   \n",
       "42780            Aerobic  Terrestrial     NaN   \n",
       "42781                NaN          NaN     NaN   \n",
       "42782                NaN          NaN     NaN   \n",
       "\n",
       "                                                comments  \\\n",
       "0      Within the Lachninae subfamily of aphids, a pu...   \n",
       "1      Sorangium cellulosum comparative genomic analysis   \n",
       "2      The genus Lysobacter consists of several plant...   \n",
       "3      Borrelia hermsii is the causative agent of tic...   \n",
       "4      confirms that the species is relatively conser...   \n",
       "...                                                  ...   \n",
       "42778  Genome sequence of Gilliamella apicola strains...   \n",
       "42779  Genome sequence of Gilliamella apicola strains...   \n",
       "42780  Whole genome sequence of P. fluorecens ATCC 17400   \n",
       "42781  To get insight into origin and evolution of St...   \n",
       "42782  A thermophilic strict anaerobe capable of cell...   \n",
       "\n",
       "                                     additional_metadata  \n",
       "0      collected_by:Armelle Coeur D'Acier\\; Emmanuell...  \n",
       "1                                                    NaN  \n",
       "2                                                    NaN  \n",
       "3                                collected_by:Tom Schwan  \n",
       "4                                                    NaN  \n",
       "...                                                  ...  \n",
       "42778                           sample_type:cell culture  \n",
       "42779                           sample_type:cell culture  \n",
       "42780                                                NaN  \n",
       "42781                         sample_type:whole organism  \n",
       "42782                                                NaN  \n",
       "\n",
       "[42783 rows x 104 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df = pd.merge(refseq_metadata_filtered, patric_metadata_filtered, left_on='#assembly_accession', right_on='assembly_accession', how='inner')\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "taxon_id\n",
       "562        58906\n",
       "1773       42792\n",
       "1313       29643\n",
       "573        28495\n",
       "1280       19675\n",
       "           ...  \n",
       "180856         1\n",
       "483218         1\n",
       "1232385        1\n",
       "429581         1\n",
       "136143         1\n",
       "Name: count, Length: 104336, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata['taxon_id'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "taxid\n",
       "562        38429\n",
       "573        21652\n",
       "1280       13493\n",
       "287        10229\n",
       "470         9529\n",
       "           ...  \n",
       "3391431        1\n",
       "3391430        1\n",
       "3391429        1\n",
       "3391428        1\n",
       "3391427        1\n",
       "Name: count, Length: 83490, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq_metadata['taxid'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "strain\n",
       "not applicable    661\n",
       "Fecal samples     634\n",
       "Not applicable     77\n",
       "not collected      67\n",
       "blood              46\n",
       "E. coli            43\n",
       "stool              33\n",
       "131                27\n",
       "10                 26\n",
       "20                 19\n",
       "15                 19\n",
       "E2                 19\n",
       "16                 18\n",
       "11                 18\n",
       "17                 17\n",
       "9                  16\n",
       "4                  16\n",
       "127                16\n",
       "ST73               16\n",
       "7                  15\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ecoli = patric_metadata[patric_metadata['taxon_id']=='562']\n",
    "ecoli['strain'].value_counts().head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "infraspecific_name\n",
       "na                             1435\n",
       "strain=ST73                      16\n",
       "strain=E2                        12\n",
       "ecotype=Quinolone resistant      12\n",
       "strain=EC2                       12\n",
       "strain=EC1                       11\n",
       "strain=EC11                      10\n",
       "strain=EC3                       10\n",
       "strain=EC14                      10\n",
       "strain=10                         9\n",
       "strain=2                          9\n",
       "strain=15                         9\n",
       "strain=EC12                       9\n",
       "strain=EC13                       9\n",
       "strain=E4                         9\n",
       "strain=EC15                       9\n",
       "strain=K-12                       8\n",
       "strain=ECO3347                    8\n",
       "strain=12                         8\n",
       "strain=EC4                        8\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ecoli = refseq_metadata[refseq_metadata['taxid']=='562']\n",
    "ecoli['infraspecific_name'].value_counts().head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>replicon_count</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>annotation_provider</th>\n",
       "      <th>annotation_name</th>\n",
       "      <th>annotation_date</th>\n",
       "      <th>total_gene_count</th>\n",
       "      <th>protein_coding_gene_count</th>\n",
       "      <th>non_coding_gene_count</th>\n",
       "      <th>pubmed_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>79721</th>\n",
       "      <td>GCF_900536975.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA104446543</td>\n",
       "      <td>UNQS00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>562</td>\n",
       "      <td>562</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>strain=ST73</td>\n",
       "      <td>2009_3</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>138</td>\n",
       "      <td>138</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_900536975.1-RS_2024_06_17</td>\n",
       "      <td>2024-06-17</td>\n",
       "      <td>5338</td>\n",
       "      <td>4876</td>\n",
       "      <td>97</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #assembly_accession   bioproject       biosample      wgs_master  \\\n",
       "79721     GCF_900536975.1  PRJNA224116  SAMEA104446543  UNQS00000000.1   \n",
       "\n",
       "      refseq_category taxid species_taxid     organism_name  \\\n",
       "79721              na   562           562  Escherichia coli   \n",
       "\n",
       "      infraspecific_name isolate  ... replicon_count scaffold_count  \\\n",
       "79721        strain=ST73  2009_3  ...              0            138   \n",
       "\n",
       "      contig_count annotation_provider                annotation_name  \\\n",
       "79721          138         NCBI RefSeq  GCF_900536975.1-RS_2024_06_17   \n",
       "\n",
       "      annotation_date total_gene_count protein_coding_gene_count  \\\n",
       "79721      2024-06-17             5338                      4876   \n",
       "\n",
       "      non_coding_gene_count pubmed_id  \n",
       "79721                    97        na  \n",
       "\n",
       "[1 rows x 38 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Step 1: Filter by taxid 562 and strain \n",
    "df_taxid_562 = refseq_metadata[refseq_metadata['taxid'] == '562']\n",
    "df_taxid_562 = df_taxid_562[df_taxid_562['infraspecific_name'] == 'strain=ST73']\n",
    "\n",
    "# Step 2: Sort by the 'annotation_date' column (ensure it's in datetime format if necessary)\n",
    "df_taxid_562['annotation_date'] = pd.to_datetime(df_taxid_562['annotation_date'])\n",
    "df_taxid_562_sorted = df_taxid_562.sort_values(by='annotation_date', ascending=False)\n",
    "\n",
    "# Step 3: Clean and convert 'genome_size' to numeric (if necessary)\n",
    "df_taxid_562_sorted['genome_size'] = df_taxid_562_sorted['genome_size'].replace({r'[^\\d.]': ''}, regex=True)\n",
    "df_taxid_562_sorted['genome_size'] = pd.to_numeric(df_taxid_562_sorted['genome_size'], errors='coerce')\n",
    "\n",
    "# Step 4: Select the latest assembly, and if there are still multiple rows, pick the one with the longest genome size\n",
    "df_taxid_562_latest = df_taxid_562_sorted.iloc[0]  # Select the latest assembly\n",
    "if len(df_taxid_562_sorted) > 1:\n",
    "    df_taxid_562_latest = df_taxid_562_sorted.nlargest(1, 'genome_size')\n",
    "\n",
    "df_taxid_562_latest\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "refseq_metadata  = refseq_metadata[refseq_metadata['infraspecific_name']!='na']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/65/rjhnld5s4xv_rsh25jvbsq5r0000gp/T/ipykernel_11308/2167343126.py:21: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.\n",
      "  df_unique_assemblies = refseq_metadata.groupby(['taxid', 'infraspecific_name'], group_keys=False).apply(select_latest_assembly)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>#assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>...</th>\n",
       "      <th>replicon_count</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>annotation_provider</th>\n",
       "      <th>annotation_name</th>\n",
       "      <th>annotation_date</th>\n",
       "      <th>total_gene_count</th>\n",
       "      <th>protein_coding_gene_count</th>\n",
       "      <th>non_coding_gene_count</th>\n",
       "      <th>pubmed_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxid</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <th>strain=DSM 101</th>\n",
       "      <td>GCF_004339465.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN10866377</td>\n",
       "      <td>SMFY00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>Ancylobacter aquaticus</td>\n",
       "      <td>strain=DSM 101</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_004339465.1-RS_2024_05_01</td>\n",
       "      <td>2024-05-01</td>\n",
       "      <td>4513</td>\n",
       "      <td>4381</td>\n",
       "      <td>53</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">100053</th>\n",
       "      <th>strain=56159</th>\n",
       "      <td>GCF_002009835.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947886</td>\n",
       "      <td>JQGT00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>strain=56159</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>337</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009835.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3803</td>\n",
       "      <td>3490</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain=56640</th>\n",
       "      <td>GCF_002009825.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947769</td>\n",
       "      <td>JQGV00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>strain=56640</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>368</td>\n",
       "      <td>368</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009825.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3815</td>\n",
       "      <td>3462</td>\n",
       "      <td>46</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain=56643</th>\n",
       "      <td>GCF_002009805.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947770</td>\n",
       "      <td>JQGU00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>strain=56643</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>335</td>\n",
       "      <td>335</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009805.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3786</td>\n",
       "      <td>3467</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain=56650</th>\n",
       "      <td>GCF_002009775.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947771</td>\n",
       "      <td>JQGS00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>strain=56650</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>341</td>\n",
       "      <td>341</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009775.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3788</td>\n",
       "      <td>3476</td>\n",
       "      <td>42</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999892</th>\n",
       "      <th>strain=P08</th>\n",
       "      <td>GCF_000204235.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02471992</td>\n",
       "      <td>AFAM00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999892</td>\n",
       "      <td>2559073</td>\n",
       "      <td>Cutibacterium modestum P08</td>\n",
       "      <td>strain=P08</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_000204235.1-RS_2024_03_28</td>\n",
       "      <td>2024-03-28</td>\n",
       "      <td>2500</td>\n",
       "      <td>2214</td>\n",
       "      <td>53</td>\n",
       "      <td>21571999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999894</th>\n",
       "      <th>strain=S95</th>\n",
       "      <td>GCF_001652585.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04623574</td>\n",
       "      <td>LWLG00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999894</td>\n",
       "      <td>999894</td>\n",
       "      <td>Thermosulfurimonas dismutans</td>\n",
       "      <td>strain=S95</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_001652585.1-RS_2024_04_21</td>\n",
       "      <td>2024-04-21</td>\n",
       "      <td>2152</td>\n",
       "      <td>2077</td>\n",
       "      <td>55</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">999931</th>\n",
       "      <th>strain=39T</th>\n",
       "      <td>GCF_910573815.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA8751438</td>\n",
       "      <td>CAJTBP000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>strain=39T</td>\n",
       "      <td>39T</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_910573815.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3390</td>\n",
       "      <td>3319</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain=DSM 24617</th>\n",
       "      <td>GCF_006716095.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN11512942</td>\n",
       "      <td>VFOK00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>strain=DSM 24617</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_006716095.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3422</td>\n",
       "      <td>3354</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain=WGS1921</th>\n",
       "      <td>GCF_043161945.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN43971778</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>strain=WGS1921</td>\n",
       "      <td>na</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_043161945.1-RS_2024_10_16</td>\n",
       "      <td>2024-10-16</td>\n",
       "      <td>3403</td>\n",
       "      <td>3308</td>\n",
       "      <td>57</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>353848 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          #assembly_accession   bioproject     biosample  \\\n",
       "taxid  infraspecific_name                                                  \n",
       "100    strain=DSM 101         GCF_004339465.1  PRJNA224116  SAMN10866377   \n",
       "100053 strain=56159           GCF_002009835.1  PRJNA224116  SAMN02947886   \n",
       "       strain=56640           GCF_002009825.1  PRJNA224116  SAMN02947769   \n",
       "       strain=56643           GCF_002009805.1  PRJNA224116  SAMN02947770   \n",
       "       strain=56650           GCF_002009775.1  PRJNA224116  SAMN02947771   \n",
       "...                                       ...          ...           ...   \n",
       "999892 strain=P08             GCF_000204235.1  PRJNA224116  SAMN02471992   \n",
       "999894 strain=S95             GCF_001652585.1  PRJNA224116  SAMN04623574   \n",
       "999931 strain=39T             GCF_910573815.1  PRJNA224116  SAMEA8751438   \n",
       "       strain=DSM 24617       GCF_006716095.1  PRJNA224116  SAMN11512942   \n",
       "       strain=WGS1921         GCF_043161945.1  PRJNA224116  SAMN43971778   \n",
       "\n",
       "                                  wgs_master   refseq_category   taxid  \\\n",
       "taxid  infraspecific_name                                                \n",
       "100    strain=DSM 101         SMFY00000000.1  reference genome     100   \n",
       "100053 strain=56159           JQGT00000000.1                na  100053   \n",
       "       strain=56640           JQGV00000000.1                na  100053   \n",
       "       strain=56643           JQGU00000000.1                na  100053   \n",
       "       strain=56650           JQGS00000000.1                na  100053   \n",
       "...                                      ...               ...     ...   \n",
       "999892 strain=P08             AFAM00000000.1  reference genome  999892   \n",
       "999894 strain=S95             LWLG00000000.1  reference genome  999894   \n",
       "999931 strain=39T          CAJTBP000000000.1                na  999931   \n",
       "       strain=DSM 24617       VFOK00000000.1  reference genome  999931   \n",
       "       strain=WGS1921                     na                na  999931   \n",
       "\n",
       "                          species_taxid                 organism_name  \\\n",
       "taxid  infraspecific_name                                               \n",
       "100    strain=DSM 101               100        Ancylobacter aquaticus   \n",
       "100053 strain=56159              100053         Leptospira alexanderi   \n",
       "       strain=56640              100053         Leptospira alexanderi   \n",
       "       strain=56643              100053         Leptospira alexanderi   \n",
       "       strain=56650              100053         Leptospira alexanderi   \n",
       "...                                 ...                           ...   \n",
       "999892 strain=P08               2559073    Cutibacterium modestum P08   \n",
       "999894 strain=S95                999894  Thermosulfurimonas dismutans   \n",
       "999931 strain=39T                999931        Barrientosiimonas humi   \n",
       "       strain=DSM 24617          999931        Barrientosiimonas humi   \n",
       "       strain=WGS1921            999931        Barrientosiimonas humi   \n",
       "\n",
       "                          infraspecific_name isolate  ... replicon_count  \\\n",
       "taxid  infraspecific_name                             ...                  \n",
       "100    strain=DSM 101         strain=DSM 101      na  ...              0   \n",
       "100053 strain=56159             strain=56159      na  ...              0   \n",
       "       strain=56640             strain=56640      na  ...              0   \n",
       "       strain=56643             strain=56643      na  ...              0   \n",
       "       strain=56650             strain=56650      na  ...              0   \n",
       "...                                      ...     ...  ...            ...   \n",
       "999892 strain=P08                 strain=P08      na  ...              0   \n",
       "999894 strain=S95                 strain=S95      na  ...              0   \n",
       "999931 strain=39T                 strain=39T     39T  ...              0   \n",
       "       strain=DSM 24617     strain=DSM 24617      na  ...              0   \n",
       "       strain=WGS1921         strain=WGS1921      na  ...              2   \n",
       "\n",
       "                          scaffold_count contig_count annotation_provider  \\\n",
       "taxid  infraspecific_name                                                   \n",
       "100    strain=DSM 101                 11           11         NCBI RefSeq   \n",
       "100053 strain=56159                  337          337         NCBI RefSeq   \n",
       "       strain=56640                  368          368         NCBI RefSeq   \n",
       "       strain=56643                  335          335         NCBI RefSeq   \n",
       "       strain=56650                  341          341         NCBI RefSeq   \n",
       "...                                  ...          ...                 ...   \n",
       "999892 strain=P08                     22           22         NCBI RefSeq   \n",
       "999894 strain=S95                     61           61         NCBI RefSeq   \n",
       "999931 strain=39T                      2            2         NCBI RefSeq   \n",
       "       strain=DSM 24617                2            2         NCBI RefSeq   \n",
       "       strain=WGS1921                  2            2         NCBI RefSeq   \n",
       "\n",
       "                                         annotation_name annotation_date  \\\n",
       "taxid  infraspecific_name                                                  \n",
       "100    strain=DSM 101      GCF_004339465.1-RS_2024_05_01      2024-05-01   \n",
       "100053 strain=56159        GCF_002009835.1-RS_2024_07_26      2024-07-26   \n",
       "       strain=56640        GCF_002009825.1-RS_2024_05_19      2024-05-19   \n",
       "       strain=56643        GCF_002009805.1-RS_2024_07_26      2024-07-26   \n",
       "       strain=56650        GCF_002009775.1-RS_2024_05_19      2024-05-19   \n",
       "...                                                  ...             ...   \n",
       "999892 strain=P08          GCF_000204235.1-RS_2024_03_28      2024-03-28   \n",
       "999894 strain=S95          GCF_001652585.1-RS_2024_04_21      2024-04-21   \n",
       "999931 strain=39T          GCF_910573815.1-RS_2024_03_29      2024-03-29   \n",
       "       strain=DSM 24617    GCF_006716095.1-RS_2024_03_29      2024-03-29   \n",
       "       strain=WGS1921      GCF_043161945.1-RS_2024_10_16      2024-10-16   \n",
       "\n",
       "                          total_gene_count protein_coding_gene_count  \\\n",
       "taxid  infraspecific_name                                              \n",
       "100    strain=DSM 101                 4513                      4381   \n",
       "100053 strain=56159                   3803                      3490   \n",
       "       strain=56640                   3815                      3462   \n",
       "       strain=56643                   3786                      3467   \n",
       "       strain=56650                   3788                      3476   \n",
       "...                                    ...                       ...   \n",
       "999892 strain=P08                     2500                      2214   \n",
       "999894 strain=S95                     2152                      2077   \n",
       "999931 strain=39T                     3390                      3319   \n",
       "       strain=DSM 24617               3422                      3354   \n",
       "       strain=WGS1921                 3403                      3308   \n",
       "\n",
       "                          non_coding_gene_count pubmed_id  \n",
       "taxid  infraspecific_name                                  \n",
       "100    strain=DSM 101                        53        na  \n",
       "100053 strain=56159                          44        na  \n",
       "       strain=56640                          46        na  \n",
       "       strain=56643                          44        na  \n",
       "       strain=56650                          42        na  \n",
       "...                                         ...       ...  \n",
       "999892 strain=P08                            53  21571999  \n",
       "999894 strain=S95                            55        na  \n",
       "999931 strain=39T                            56        na  \n",
       "       strain=DSM 24617                      56        na  \n",
       "       strain=WGS1921                        57        na  \n",
       "\n",
       "[353848 rows x 38 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming 'refseq_metadata' is your DataFrame\n",
    "def select_latest_assembly(group):\n",
    "    # Sort by 'annotation_date' and convert to datetime if necessary\n",
    "    group['annotation_date'] = pd.to_datetime(group['annotation_date'])\n",
    "    group_sorted = group.sort_values(by='annotation_date', ascending=False)\n",
    "    \n",
    "    # Clean and convert 'genome_size' to numeric\n",
    "    group_sorted['genome_size'] = group_sorted['genome_size'].replace({r'[^\\d.]': ''}, regex=True)\n",
    "    group_sorted['genome_size'] = pd.to_numeric(group_sorted['genome_size'], errors='coerce')\n",
    "    \n",
    "    # Select the latest assembly, and if there are still multiple rows, pick the one with the longest genome size\n",
    "    latest_assembly = group_sorted.iloc[0]  # Select the latest assembly\n",
    "    if len(group_sorted) > 1:\n",
    "        latest_assembly = group_sorted.nlargest(1, 'genome_size').iloc[0]\n",
    "    \n",
    "    return latest_assembly\n",
    "\n",
    "# Apply the function to each group (grouped by 'taxid' and 'infraspecific_name')\n",
    "df_unique_assemblies = refseq_metadata.groupby(['taxid', 'infraspecific_name'], group_keys=False).apply(select_latest_assembly)\n",
    "\n",
    "# The result will contain the unique assembly for each strain\n",
    "df_unique_assemblies\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "infraspecific_name\n",
       "strain=BIOML-A1      71\n",
       "strain=BIOML-A2      53\n",
       "strain=S1            49\n",
       "strain=BIOML-A3      49\n",
       "strain=C1            47\n",
       "                     ..\n",
       "strain=CCUG 60898     1\n",
       "strain=TA208          1\n",
       "strain=39T            1\n",
       "strain=DSM 24617      1\n",
       "strain=WGS1921        1\n",
       "Name: count, Length: 329564, dtype: int64"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_unique_assemblies['infraspecific_name'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unique_assemblies.to_csv(\"refseq_representative_assemblies.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "strain\n",
       "B31 5A4                   966\n",
       "CO275                     769\n",
       "Neisseria meningitidis    542\n",
       "B31 5A3                   439\n",
       "5A18NP1                   284\n",
       "                         ... \n",
       "YR_bin-0414                 1\n",
       "YR_bin-2355                 1\n",
       "N43453                      1\n",
       "N43451                      1\n",
       "N43450                      1\n",
       "Name: count, Length: 878075, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata['strain'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "patric_metadata = patric_metadata[patric_metadata['strain']!='not applicable']\n",
    "patric_metadata = patric_metadata[patric_metadata['strain']!='Not applicable']\n",
    "patric_metadata = patric_metadata[patric_metadata['strain']!='Fecal samples']\n",
    "patric_metadata = patric_metadata[patric_metadata['strain']!='not collected']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['genome_id', 'genome_name', 'organism_name', 'taxon_id',\n",
       "       'genome_status', 'strain', 'serovar', 'biovar', 'pathovar', 'mlst',\n",
       "       'other_typing', 'culture_collection', 'type_strain', 'completion_date',\n",
       "       'publication', 'bioproject_accession', 'biosample_accession',\n",
       "       'assembly_accession', 'genbank_accessions', 'refseq_accessions',\n",
       "       'sequencing_centers', 'sequencing_status', 'sequencing_platform',\n",
       "       'sequencing_depth', 'assembly_method', 'chromosomes', 'plasmids',\n",
       "       'contigs', 'sequences', 'genome_length', 'gc_content', 'patric_cds',\n",
       "       'brc1_cds', 'refseq_cds', 'isolation_site', 'isolation_source',\n",
       "       'isolation_comments', 'collection_date', 'isolation_country',\n",
       "       'geographic_location', 'latitude', 'longitude', 'altitude', 'depth',\n",
       "       'other_environmental', 'host_name', 'host_gender', 'host_age',\n",
       "       'host_health', 'body_sample_site', 'body_sample_subsite',\n",
       "       'other_clinical', 'antimicrobial_resistance',\n",
       "       'antimicrobial_resistance_evidence', 'gram_stain', 'cell_shape',\n",
       "       'motility', 'sporulation', 'temperature_range', 'optimal_temperature',\n",
       "       'salinity', 'oxygen_requirement', 'habitat', 'disease', 'comments',\n",
       "       'additional_metadata'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patric_metadata.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unique_assemblies_patric.to_csv(\"patric_representative_assemblies.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# Assuming 'refseq_metadata' is your DataFrame\n",
    "def select_latest_assembly(group):\n",
    "    # Sort by 'annotation_date' and convert to datetime if necessary\n",
    "    group['completion_date'] = pd.to_datetime(group['completion_date'])\n",
    "    group_sorted = group.sort_values(by='completion_date', ascending=False)\n",
    "    \n",
    "    # Clean and convert 'genome_size' to numeric\n",
    "    group_sorted['genome_length'] = group_sorted['genome_length'].replace({r'[^\\d.]': ''}, regex=True)\n",
    "    group_sorted['genome_length'] = pd.to_numeric(group_sorted['genome_length'], errors='coerce')\n",
    "    \n",
    "    # Select the latest assembly, and if there are still multiple rows, pick the one with the longest genome size\n",
    "    latest_assembly = group_sorted.iloc[0]  # Select the latest assembly\n",
    "    if len(group_sorted) > 1:\n",
    "        latest_assembly = group_sorted.nlargest(1, 'genome_length').iloc[0]\n",
    "    \n",
    "    return latest_assembly\n",
    "\n",
    "# Apply the function to each group (grouped by 'taxid' and 'infraspecific_name')\n",
    "df_unique_assemblies_patric = patric_metadata.groupby(['taxon_id', 'strain'], group_keys=False).apply(select_latest_assembly)\n",
    "\n",
    "# The result will contain the unique assembly for each strain\n",
    "df_unique_assemblies_patric\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>genome_id</th>\n",
       "      <th>genome_name</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>genome_status</th>\n",
       "      <th>strain</th>\n",
       "      <th>serovar</th>\n",
       "      <th>biovar</th>\n",
       "      <th>pathovar</th>\n",
       "      <th>mlst</th>\n",
       "      <th>...</th>\n",
       "      <th>motility</th>\n",
       "      <th>sporulation</th>\n",
       "      <th>temperature_range</th>\n",
       "      <th>optimal_temperature</th>\n",
       "      <th>salinity</th>\n",
       "      <th>oxygen_requirement</th>\n",
       "      <th>habitat</th>\n",
       "      <th>disease</th>\n",
       "      <th>comments</th>\n",
       "      <th>additional_metadata</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">100</th>\n",
       "      <th>DSM 101</th>\n",
       "      <td>100.11</td>\n",
       "      <td>Ancylobacter aquaticus strain DSM 101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100</td>\n",
       "      <td>WGS</td>\n",
       "      <td>DSM 101</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Genome sequencing as part of The One Thousand ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UV5</th>\n",
       "      <td>100.9</td>\n",
       "      <td>Ancylobacter aquaticus strain UV5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100</td>\n",
       "      <td>WGS</td>\n",
       "      <td>UV5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Ancylobacter aquaticus sp. strain UV5, isolate...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">100053</th>\n",
       "      <th>56159</th>\n",
       "      <td>100053.7</td>\n",
       "      <td>Leptospira alexanderi strain 56159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100053</td>\n",
       "      <td>WGS</td>\n",
       "      <td>56159</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Comparative genomics analysis of 102 Leptospir...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56640</th>\n",
       "      <td>100053.6</td>\n",
       "      <td>Leptospira alexanderi strain 56640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100053</td>\n",
       "      <td>WGS</td>\n",
       "      <td>56640</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Comparative genomics analysis of 102 Leptospir...</td>\n",
       "      <td>collected_by:Jincai Qin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56643</th>\n",
       "      <td>100053.5</td>\n",
       "      <td>Leptospira alexanderi strain 56643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>100053</td>\n",
       "      <td>WGS</td>\n",
       "      <td>56643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Comparative genomics analysis of 102 Leptospir...</td>\n",
       "      <td>collected_by:Jincai Qin</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">999894</th>\n",
       "      <th>HyVt-483</th>\n",
       "      <td>999894.10</td>\n",
       "      <td>Thermosulfurimonas dismutans strain HyVt-483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>999894</td>\n",
       "      <td>WGS</td>\n",
       "      <td>HyVt-483</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This is the collection of metagenome-assembled...</td>\n",
       "      <td>sample_type:metagenomic assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S95</th>\n",
       "      <td>999894.9</td>\n",
       "      <td>Thermosulfurimonas dismutans strain S95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>999894</td>\n",
       "      <td>WGS</td>\n",
       "      <td>S95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The genome sequncing and analysis of Thermosul...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999898</th>\n",
       "      <th>CEB3</th>\n",
       "      <td>999898.5</td>\n",
       "      <td>Peptococcaceae bacterium CEB3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>999898</td>\n",
       "      <td>WGS</td>\n",
       "      <td>CEB3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Strain CEB3 is an acidophilic sulfate reducing...</td>\n",
       "      <td>sample_type:whole organism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">999931</th>\n",
       "      <th>39T</th>\n",
       "      <td>999931.5</td>\n",
       "      <td>Barrientosiimonas humi strain type strain: 39 ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>999931</td>\n",
       "      <td>WGS</td>\n",
       "      <td>39T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A genomic analysis study presents the whole-ge...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DSM 24617</th>\n",
       "      <td>999931.3</td>\n",
       "      <td>Barrientosiimonas humi strain DSM 24617</td>\n",
       "      <td>NaN</td>\n",
       "      <td>999931</td>\n",
       "      <td>WGS</td>\n",
       "      <td>DSM 24617</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Sequencing of Actinobacteria genome as a plant...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>912435 rows × 66 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    genome_id  \\\n",
       "taxon_id strain                 \n",
       "100      DSM 101       100.11   \n",
       "         UV5            100.9   \n",
       "100053   56159       100053.7   \n",
       "         56640       100053.6   \n",
       "         56643       100053.5   \n",
       "...                       ...   \n",
       "999894   HyVt-483   999894.10   \n",
       "         S95         999894.9   \n",
       "999898   CEB3        999898.5   \n",
       "999931   39T         999931.5   \n",
       "         DSM 24617   999931.3   \n",
       "\n",
       "                                                          genome_name  \\\n",
       "taxon_id strain                                                         \n",
       "100      DSM 101                Ancylobacter aquaticus strain DSM 101   \n",
       "         UV5                        Ancylobacter aquaticus strain UV5   \n",
       "100053   56159                     Leptospira alexanderi strain 56159   \n",
       "         56640                     Leptospira alexanderi strain 56640   \n",
       "         56643                     Leptospira alexanderi strain 56643   \n",
       "...                                                               ...   \n",
       "999894   HyVt-483        Thermosulfurimonas dismutans strain HyVt-483   \n",
       "         S95                  Thermosulfurimonas dismutans strain S95   \n",
       "999898   CEB3                           Peptococcaceae bacterium CEB3   \n",
       "999931   39T        Barrientosiimonas humi strain type strain: 39 ...   \n",
       "         DSM 24617            Barrientosiimonas humi strain DSM 24617   \n",
       "\n",
       "                   organism_name taxon_id genome_status     strain serovar  \\\n",
       "taxon_id strain                                                              \n",
       "100      DSM 101             NaN      100           WGS    DSM 101     NaN   \n",
       "         UV5                 NaN      100           WGS        UV5     NaN   \n",
       "100053   56159               NaN   100053           WGS      56159     NaN   \n",
       "         56640               NaN   100053           WGS      56640     NaN   \n",
       "         56643               NaN   100053           WGS      56643     NaN   \n",
       "...                          ...      ...           ...        ...     ...   \n",
       "999894   HyVt-483            NaN   999894           WGS   HyVt-483     NaN   \n",
       "         S95                 NaN   999894           WGS        S95     NaN   \n",
       "999898   CEB3                NaN   999898           WGS       CEB3     NaN   \n",
       "999931   39T                 NaN   999931           WGS        39T     NaN   \n",
       "         DSM 24617           NaN   999931           WGS  DSM 24617     NaN   \n",
       "\n",
       "                   biovar pathovar mlst  ... motility sporulation  \\\n",
       "taxon_id strain                          ...                        \n",
       "100      DSM 101      NaN      NaN  NaN  ...      NaN         NaN   \n",
       "         UV5          NaN      NaN  NaN  ...      NaN         NaN   \n",
       "100053   56159        NaN      NaN  NaN  ...      NaN         NaN   \n",
       "         56640        NaN      NaN  NaN  ...      NaN         NaN   \n",
       "         56643        NaN      NaN  NaN  ...      NaN         NaN   \n",
       "...                   ...      ...  ...  ...      ...         ...   \n",
       "999894   HyVt-483     NaN      NaN  NaN  ...      NaN         NaN   \n",
       "         S95          NaN      NaN  NaN  ...      NaN         NaN   \n",
       "999898   CEB3         NaN      NaN  NaN  ...      NaN         NaN   \n",
       "999931   39T          NaN      NaN  NaN  ...      NaN         NaN   \n",
       "         DSM 24617    NaN      NaN  NaN  ...      NaN         NaN   \n",
       "\n",
       "                   temperature_range optimal_temperature salinity  \\\n",
       "taxon_id strain                                                     \n",
       "100      DSM 101                 NaN                 NaN      NaN   \n",
       "         UV5                     NaN                 NaN      NaN   \n",
       "100053   56159                   NaN                 NaN      NaN   \n",
       "         56640                   NaN                 NaN      NaN   \n",
       "         56643                   NaN                 NaN      NaN   \n",
       "...                              ...                 ...      ...   \n",
       "999894   HyVt-483                NaN                 NaN      NaN   \n",
       "         S95                     NaN                 NaN      NaN   \n",
       "999898   CEB3                    NaN                   C      NaN   \n",
       "999931   39T                     NaN                 NaN      NaN   \n",
       "         DSM 24617               NaN                 NaN      NaN   \n",
       "\n",
       "                   oxygen_requirement habitat disease  \\\n",
       "taxon_id strain                                         \n",
       "100      DSM 101                  NaN     NaN     NaN   \n",
       "         UV5                      NaN     NaN     NaN   \n",
       "100053   56159                    NaN     NaN     NaN   \n",
       "         56640                    NaN     NaN     NaN   \n",
       "         56643                    NaN     NaN     NaN   \n",
       "...                               ...     ...     ...   \n",
       "999894   HyVt-483                 NaN     NaN     NaN   \n",
       "         S95                      NaN     NaN     NaN   \n",
       "999898   CEB3                     NaN     NaN     NaN   \n",
       "999931   39T                      NaN     NaN     NaN   \n",
       "         DSM 24617                NaN     NaN     NaN   \n",
       "\n",
       "                                                             comments  \\\n",
       "taxon_id strain                                                         \n",
       "100      DSM 101    Genome sequencing as part of The One Thousand ...   \n",
       "         UV5        Ancylobacter aquaticus sp. strain UV5, isolate...   \n",
       "100053   56159      Comparative genomics analysis of 102 Leptospir...   \n",
       "         56640      Comparative genomics analysis of 102 Leptospir...   \n",
       "         56643      Comparative genomics analysis of 102 Leptospir...   \n",
       "...                                                               ...   \n",
       "999894   HyVt-483   This is the collection of metagenome-assembled...   \n",
       "         S95        The genome sequncing and analysis of Thermosul...   \n",
       "999898   CEB3       Strain CEB3 is an acidophilic sulfate reducing...   \n",
       "999931   39T        A genomic analysis study presents the whole-ge...   \n",
       "         DSM 24617  Sequencing of Actinobacteria genome as a plant...   \n",
       "\n",
       "                                 additional_metadata  \n",
       "taxon_id strain                                       \n",
       "100      DSM 101                                 NaN  \n",
       "         UV5                                     NaN  \n",
       "100053   56159                                   NaN  \n",
       "         56640               collected_by:Jincai Qin  \n",
       "         56643               collected_by:Jincai Qin  \n",
       "...                                              ...  \n",
       "999894   HyVt-483   sample_type:metagenomic assembly  \n",
       "         S95                                     NaN  \n",
       "999898   CEB3             sample_type:whole organism  \n",
       "999931   39T                                     NaN  \n",
       "         DSM 24617                               NaN  \n",
       "\n",
       "[912435 rows x 66 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_unique_assemblies_patric"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unique_assemblies_refseq = pd.read_csv(\"refseq_representative_assemblies.csv\",dtype=str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_unique_assemblies_patric = pd.read_csv(\"patric_representative_assemblies.csv\",dtype=str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th>#assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid.1</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>...</th>\n",
       "      <th>replicon_count</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>annotation_provider</th>\n",
       "      <th>annotation_name</th>\n",
       "      <th>annotation_date</th>\n",
       "      <th>total_gene_count</th>\n",
       "      <th>protein_coding_gene_count</th>\n",
       "      <th>non_coding_gene_count</th>\n",
       "      <th>pubmed_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>strain=DSM 101</td>\n",
       "      <td>GCF_004339465.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN10866377</td>\n",
       "      <td>SMFY00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>Ancylobacter aquaticus</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_004339465.1-RS_2024_05_01</td>\n",
       "      <td>2024-05-01</td>\n",
       "      <td>4513</td>\n",
       "      <td>4381</td>\n",
       "      <td>53</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100053</td>\n",
       "      <td>strain=56159</td>\n",
       "      <td>GCF_002009835.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947886</td>\n",
       "      <td>JQGT00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>337</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009835.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3803</td>\n",
       "      <td>3490</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100053</td>\n",
       "      <td>strain=56640</td>\n",
       "      <td>GCF_002009825.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947769</td>\n",
       "      <td>JQGV00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>368</td>\n",
       "      <td>368</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009825.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3815</td>\n",
       "      <td>3462</td>\n",
       "      <td>46</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100053</td>\n",
       "      <td>strain=56643</td>\n",
       "      <td>GCF_002009805.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947770</td>\n",
       "      <td>JQGU00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>335</td>\n",
       "      <td>335</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009805.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3786</td>\n",
       "      <td>3467</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100053</td>\n",
       "      <td>strain=56650</td>\n",
       "      <td>GCF_002009775.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947771</td>\n",
       "      <td>JQGS00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>341</td>\n",
       "      <td>341</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009775.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3788</td>\n",
       "      <td>3476</td>\n",
       "      <td>42</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353843</th>\n",
       "      <td>999892</td>\n",
       "      <td>strain=P08</td>\n",
       "      <td>GCF_000204235.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02471992</td>\n",
       "      <td>AFAM00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999892</td>\n",
       "      <td>2559073</td>\n",
       "      <td>Cutibacterium modestum P08</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_000204235.1-RS_2024_03_28</td>\n",
       "      <td>2024-03-28</td>\n",
       "      <td>2500</td>\n",
       "      <td>2214</td>\n",
       "      <td>53</td>\n",
       "      <td>21571999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353844</th>\n",
       "      <td>999894</td>\n",
       "      <td>strain=S95</td>\n",
       "      <td>GCF_001652585.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04623574</td>\n",
       "      <td>LWLG00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999894</td>\n",
       "      <td>999894</td>\n",
       "      <td>Thermosulfurimonas dismutans</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_001652585.1-RS_2024_04_21</td>\n",
       "      <td>2024-04-21</td>\n",
       "      <td>2152</td>\n",
       "      <td>2077</td>\n",
       "      <td>55</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353845</th>\n",
       "      <td>999931</td>\n",
       "      <td>strain=39T</td>\n",
       "      <td>GCF_910573815.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA8751438</td>\n",
       "      <td>CAJTBP000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_910573815.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3390</td>\n",
       "      <td>3319</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353846</th>\n",
       "      <td>999931</td>\n",
       "      <td>strain=DSM 24617</td>\n",
       "      <td>GCF_006716095.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN11512942</td>\n",
       "      <td>VFOK00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_006716095.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3422</td>\n",
       "      <td>3354</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353847</th>\n",
       "      <td>999931</td>\n",
       "      <td>strain=WGS1921</td>\n",
       "      <td>GCF_043161945.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN43971778</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_043161945.1-RS_2024_10_16</td>\n",
       "      <td>2024-10-16</td>\n",
       "      <td>3403</td>\n",
       "      <td>3308</td>\n",
       "      <td>57</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>353848 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id            strain #assembly_accession   bioproject  \\\n",
       "0           100    strain=DSM 101     GCF_004339465.1  PRJNA224116   \n",
       "1        100053      strain=56159     GCF_002009835.1  PRJNA224116   \n",
       "2        100053      strain=56640     GCF_002009825.1  PRJNA224116   \n",
       "3        100053      strain=56643     GCF_002009805.1  PRJNA224116   \n",
       "4        100053      strain=56650     GCF_002009775.1  PRJNA224116   \n",
       "...         ...               ...                 ...          ...   \n",
       "353843   999892        strain=P08     GCF_000204235.1  PRJNA224116   \n",
       "353844   999894        strain=S95     GCF_001652585.1  PRJNA224116   \n",
       "353845   999931        strain=39T     GCF_910573815.1  PRJNA224116   \n",
       "353846   999931  strain=DSM 24617     GCF_006716095.1  PRJNA224116   \n",
       "353847   999931    strain=WGS1921     GCF_043161945.1  PRJNA224116   \n",
       "\n",
       "           biosample         wgs_master   refseq_category taxid.1  \\\n",
       "0       SAMN10866377     SMFY00000000.1  reference genome     100   \n",
       "1       SAMN02947886     JQGT00000000.1                na  100053   \n",
       "2       SAMN02947769     JQGV00000000.1                na  100053   \n",
       "3       SAMN02947770     JQGU00000000.1                na  100053   \n",
       "4       SAMN02947771     JQGS00000000.1                na  100053   \n",
       "...              ...                ...               ...     ...   \n",
       "353843  SAMN02471992     AFAM00000000.1  reference genome  999892   \n",
       "353844  SAMN04623574     LWLG00000000.1  reference genome  999894   \n",
       "353845  SAMEA8751438  CAJTBP000000000.1                na  999931   \n",
       "353846  SAMN11512942     VFOK00000000.1  reference genome  999931   \n",
       "353847  SAMN43971778                 na                na  999931   \n",
       "\n",
       "       species_taxid                 organism_name  ... replicon_count  \\\n",
       "0                100        Ancylobacter aquaticus  ...              0   \n",
       "1             100053         Leptospira alexanderi  ...              0   \n",
       "2             100053         Leptospira alexanderi  ...              0   \n",
       "3             100053         Leptospira alexanderi  ...              0   \n",
       "4             100053         Leptospira alexanderi  ...              0   \n",
       "...              ...                           ...  ...            ...   \n",
       "353843       2559073    Cutibacterium modestum P08  ...              0   \n",
       "353844        999894  Thermosulfurimonas dismutans  ...              0   \n",
       "353845        999931        Barrientosiimonas humi  ...              0   \n",
       "353846        999931        Barrientosiimonas humi  ...              0   \n",
       "353847        999931        Barrientosiimonas humi  ...              2   \n",
       "\n",
       "       scaffold_count contig_count annotation_provider  \\\n",
       "0                  11           11         NCBI RefSeq   \n",
       "1                 337          337         NCBI RefSeq   \n",
       "2                 368          368         NCBI RefSeq   \n",
       "3                 335          335         NCBI RefSeq   \n",
       "4                 341          341         NCBI RefSeq   \n",
       "...               ...          ...                 ...   \n",
       "353843             22           22         NCBI RefSeq   \n",
       "353844             61           61         NCBI RefSeq   \n",
       "353845              2            2         NCBI RefSeq   \n",
       "353846              2            2         NCBI RefSeq   \n",
       "353847              2            2         NCBI RefSeq   \n",
       "\n",
       "                      annotation_name annotation_date total_gene_count  \\\n",
       "0       GCF_004339465.1-RS_2024_05_01      2024-05-01             4513   \n",
       "1       GCF_002009835.1-RS_2024_07_26      2024-07-26             3803   \n",
       "2       GCF_002009825.1-RS_2024_05_19      2024-05-19             3815   \n",
       "3       GCF_002009805.1-RS_2024_07_26      2024-07-26             3786   \n",
       "4       GCF_002009775.1-RS_2024_05_19      2024-05-19             3788   \n",
       "...                               ...             ...              ...   \n",
       "353843  GCF_000204235.1-RS_2024_03_28      2024-03-28             2500   \n",
       "353844  GCF_001652585.1-RS_2024_04_21      2024-04-21             2152   \n",
       "353845  GCF_910573815.1-RS_2024_03_29      2024-03-29             3390   \n",
       "353846  GCF_006716095.1-RS_2024_03_29      2024-03-29             3422   \n",
       "353847  GCF_043161945.1-RS_2024_10_16      2024-10-16             3403   \n",
       "\n",
       "       protein_coding_gene_count non_coding_gene_count pubmed_id  \n",
       "0                           4381                    53        na  \n",
       "1                           3490                    44        na  \n",
       "2                           3462                    46        na  \n",
       "3                           3467                    44        na  \n",
       "4                           3476                    42        na  \n",
       "...                          ...                   ...       ...  \n",
       "353843                      2214                    53  21571999  \n",
       "353844                      2077                    55        na  \n",
       "353845                      3319                    56        na  \n",
       "353846                      3354                    56        na  \n",
       "353847                      3308                    57        na  \n",
       "\n",
       "[353848 rows x 40 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename columns in the final result\n",
    "df_unique_assemblies_refseq = df_unique_assemblies_refseq.rename(columns={'taxid': 'taxon_id', 'infraspecific_name': 'strain'})\n",
    "\n",
    "# The result will now have 'taxon_id' and 'strain' as column names\n",
    "df_unique_assemblies_refseq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th>#assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid.1</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>...</th>\n",
       "      <th>replicon_count</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>annotation_provider</th>\n",
       "      <th>annotation_name</th>\n",
       "      <th>annotation_date</th>\n",
       "      <th>total_gene_count</th>\n",
       "      <th>protein_coding_gene_count</th>\n",
       "      <th>non_coding_gene_count</th>\n",
       "      <th>pubmed_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>DSM 101</td>\n",
       "      <td>GCF_004339465.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN10866377</td>\n",
       "      <td>SMFY00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>Ancylobacter aquaticus</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_004339465.1-RS_2024_05_01</td>\n",
       "      <td>2024-05-01</td>\n",
       "      <td>4513</td>\n",
       "      <td>4381</td>\n",
       "      <td>53</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100053</td>\n",
       "      <td>56159</td>\n",
       "      <td>GCF_002009835.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947886</td>\n",
       "      <td>JQGT00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>337</td>\n",
       "      <td>337</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009835.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3803</td>\n",
       "      <td>3490</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100053</td>\n",
       "      <td>56640</td>\n",
       "      <td>GCF_002009825.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947769</td>\n",
       "      <td>JQGV00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>368</td>\n",
       "      <td>368</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009825.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3815</td>\n",
       "      <td>3462</td>\n",
       "      <td>46</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100053</td>\n",
       "      <td>56643</td>\n",
       "      <td>GCF_002009805.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947770</td>\n",
       "      <td>JQGU00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>335</td>\n",
       "      <td>335</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009805.1-RS_2024_07_26</td>\n",
       "      <td>2024-07-26</td>\n",
       "      <td>3786</td>\n",
       "      <td>3467</td>\n",
       "      <td>44</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100053</td>\n",
       "      <td>56650</td>\n",
       "      <td>GCF_002009775.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02947771</td>\n",
       "      <td>JQGS00000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>100053</td>\n",
       "      <td>100053</td>\n",
       "      <td>Leptospira alexanderi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>341</td>\n",
       "      <td>341</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_002009775.1-RS_2024_05_19</td>\n",
       "      <td>2024-05-19</td>\n",
       "      <td>3788</td>\n",
       "      <td>3476</td>\n",
       "      <td>42</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353843</th>\n",
       "      <td>999892</td>\n",
       "      <td>P08</td>\n",
       "      <td>GCF_000204235.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN02471992</td>\n",
       "      <td>AFAM00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999892</td>\n",
       "      <td>2559073</td>\n",
       "      <td>Cutibacterium modestum P08</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_000204235.1-RS_2024_03_28</td>\n",
       "      <td>2024-03-28</td>\n",
       "      <td>2500</td>\n",
       "      <td>2214</td>\n",
       "      <td>53</td>\n",
       "      <td>21571999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353844</th>\n",
       "      <td>999894</td>\n",
       "      <td>S95</td>\n",
       "      <td>GCF_001652585.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN04623574</td>\n",
       "      <td>LWLG00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999894</td>\n",
       "      <td>999894</td>\n",
       "      <td>Thermosulfurimonas dismutans</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>61</td>\n",
       "      <td>61</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_001652585.1-RS_2024_04_21</td>\n",
       "      <td>2024-04-21</td>\n",
       "      <td>2152</td>\n",
       "      <td>2077</td>\n",
       "      <td>55</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353845</th>\n",
       "      <td>999931</td>\n",
       "      <td>39T</td>\n",
       "      <td>GCF_910573815.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMEA8751438</td>\n",
       "      <td>CAJTBP000000000.1</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_910573815.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3390</td>\n",
       "      <td>3319</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353846</th>\n",
       "      <td>999931</td>\n",
       "      <td>DSM 24617</td>\n",
       "      <td>GCF_006716095.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN11512942</td>\n",
       "      <td>VFOK00000000.1</td>\n",
       "      <td>reference genome</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_006716095.1-RS_2024_03_29</td>\n",
       "      <td>2024-03-29</td>\n",
       "      <td>3422</td>\n",
       "      <td>3354</td>\n",
       "      <td>56</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353847</th>\n",
       "      <td>999931</td>\n",
       "      <td>WGS1921</td>\n",
       "      <td>GCF_043161945.1</td>\n",
       "      <td>PRJNA224116</td>\n",
       "      <td>SAMN43971778</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>999931</td>\n",
       "      <td>999931</td>\n",
       "      <td>Barrientosiimonas humi</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>NCBI RefSeq</td>\n",
       "      <td>GCF_043161945.1-RS_2024_10_16</td>\n",
       "      <td>2024-10-16</td>\n",
       "      <td>3403</td>\n",
       "      <td>3308</td>\n",
       "      <td>57</td>\n",
       "      <td>na</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>353848 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id     strain #assembly_accession   bioproject     biosample  \\\n",
       "0           100    DSM 101     GCF_004339465.1  PRJNA224116  SAMN10866377   \n",
       "1        100053      56159     GCF_002009835.1  PRJNA224116  SAMN02947886   \n",
       "2        100053      56640     GCF_002009825.1  PRJNA224116  SAMN02947769   \n",
       "3        100053      56643     GCF_002009805.1  PRJNA224116  SAMN02947770   \n",
       "4        100053      56650     GCF_002009775.1  PRJNA224116  SAMN02947771   \n",
       "...         ...        ...                 ...          ...           ...   \n",
       "353843   999892        P08     GCF_000204235.1  PRJNA224116  SAMN02471992   \n",
       "353844   999894        S95     GCF_001652585.1  PRJNA224116  SAMN04623574   \n",
       "353845   999931        39T     GCF_910573815.1  PRJNA224116  SAMEA8751438   \n",
       "353846   999931  DSM 24617     GCF_006716095.1  PRJNA224116  SAMN11512942   \n",
       "353847   999931    WGS1921     GCF_043161945.1  PRJNA224116  SAMN43971778   \n",
       "\n",
       "               wgs_master   refseq_category taxid.1 species_taxid  \\\n",
       "0          SMFY00000000.1  reference genome     100           100   \n",
       "1          JQGT00000000.1                na  100053        100053   \n",
       "2          JQGV00000000.1                na  100053        100053   \n",
       "3          JQGU00000000.1                na  100053        100053   \n",
       "4          JQGS00000000.1                na  100053        100053   \n",
       "...                   ...               ...     ...           ...   \n",
       "353843     AFAM00000000.1  reference genome  999892       2559073   \n",
       "353844     LWLG00000000.1  reference genome  999894        999894   \n",
       "353845  CAJTBP000000000.1                na  999931        999931   \n",
       "353846     VFOK00000000.1  reference genome  999931        999931   \n",
       "353847                 na                na  999931        999931   \n",
       "\n",
       "                       organism_name  ... replicon_count scaffold_count  \\\n",
       "0             Ancylobacter aquaticus  ...              0             11   \n",
       "1              Leptospira alexanderi  ...              0            337   \n",
       "2              Leptospira alexanderi  ...              0            368   \n",
       "3              Leptospira alexanderi  ...              0            335   \n",
       "4              Leptospira alexanderi  ...              0            341   \n",
       "...                              ...  ...            ...            ...   \n",
       "353843    Cutibacterium modestum P08  ...              0             22   \n",
       "353844  Thermosulfurimonas dismutans  ...              0             61   \n",
       "353845        Barrientosiimonas humi  ...              0              2   \n",
       "353846        Barrientosiimonas humi  ...              0              2   \n",
       "353847        Barrientosiimonas humi  ...              2              2   \n",
       "\n",
       "       contig_count annotation_provider                annotation_name  \\\n",
       "0                11         NCBI RefSeq  GCF_004339465.1-RS_2024_05_01   \n",
       "1               337         NCBI RefSeq  GCF_002009835.1-RS_2024_07_26   \n",
       "2               368         NCBI RefSeq  GCF_002009825.1-RS_2024_05_19   \n",
       "3               335         NCBI RefSeq  GCF_002009805.1-RS_2024_07_26   \n",
       "4               341         NCBI RefSeq  GCF_002009775.1-RS_2024_05_19   \n",
       "...             ...                 ...                            ...   \n",
       "353843           22         NCBI RefSeq  GCF_000204235.1-RS_2024_03_28   \n",
       "353844           61         NCBI RefSeq  GCF_001652585.1-RS_2024_04_21   \n",
       "353845            2         NCBI RefSeq  GCF_910573815.1-RS_2024_03_29   \n",
       "353846            2         NCBI RefSeq  GCF_006716095.1-RS_2024_03_29   \n",
       "353847            2         NCBI RefSeq  GCF_043161945.1-RS_2024_10_16   \n",
       "\n",
       "       annotation_date total_gene_count protein_coding_gene_count  \\\n",
       "0           2024-05-01             4513                      4381   \n",
       "1           2024-07-26             3803                      3490   \n",
       "2           2024-05-19             3815                      3462   \n",
       "3           2024-07-26             3786                      3467   \n",
       "4           2024-05-19             3788                      3476   \n",
       "...                ...              ...                       ...   \n",
       "353843      2024-03-28             2500                      2214   \n",
       "353844      2024-04-21             2152                      2077   \n",
       "353845      2024-03-29             3390                      3319   \n",
       "353846      2024-03-29             3422                      3354   \n",
       "353847      2024-10-16             3403                      3308   \n",
       "\n",
       "       non_coding_gene_count pubmed_id  \n",
       "0                         53        na  \n",
       "1                         44        na  \n",
       "2                         46        na  \n",
       "3                         44        na  \n",
       "4                         42        na  \n",
       "...                      ...       ...  \n",
       "353843                    53  21571999  \n",
       "353844                    55        na  \n",
       "353845                    56        na  \n",
       "353846                    56        na  \n",
       "353847                    57        na  \n",
       "\n",
       "[353848 rows x 40 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove \"strain=\" from every value in the 'strain' column\n",
    "df_unique_assemblies_refseq['strain'] = df_unique_assemblies_refseq['strain'].str.replace('strain=', '', regex=False)\n",
    "\n",
    "# The result will have \"strain=\" removed from the 'strain' column\n",
    "df_unique_assemblies_refseq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['taxon_id', 'strain', 'genome_id', 'genome_name', 'organism_name',\n",
       "       'taxon_id.1', 'genome_status', 'strain.1', 'serovar', 'biovar',\n",
       "       'pathovar', 'mlst', 'other_typing', 'culture_collection', 'type_strain',\n",
       "       'completion_date', 'publication', 'bioproject_accession',\n",
       "       'biosample_accession', 'assembly_accession', 'genbank_accessions',\n",
       "       'refseq_accessions', 'sequencing_centers', 'sequencing_status',\n",
       "       'sequencing_platform', 'sequencing_depth', 'assembly_method',\n",
       "       'chromosomes', 'plasmids', 'contigs', 'sequences', 'genome_length',\n",
       "       'gc_content', 'patric_cds', 'brc1_cds', 'refseq_cds', 'isolation_site',\n",
       "       'isolation_source', 'isolation_comments', 'collection_date',\n",
       "       'isolation_country', 'geographic_location', 'latitude', 'longitude',\n",
       "       'altitude', 'depth', 'other_environmental', 'host_name', 'host_gender',\n",
       "       'host_age', 'host_health', 'body_sample_site', 'body_sample_subsite',\n",
       "       'other_clinical', 'antimicrobial_resistance',\n",
       "       'antimicrobial_resistance_evidence', 'gram_stain', 'cell_shape',\n",
       "       'motility', 'sporulation', 'temperature_range', 'optimal_temperature',\n",
       "       'salinity', 'oxygen_requirement', 'habitat', 'disease', 'comments',\n",
       "       'additional_metadata'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_unique_assemblies_patric.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>DSM 101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100053</td>\n",
       "      <td>56159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100053</td>\n",
       "      <td>56640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100053</td>\n",
       "      <td>56643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100053</td>\n",
       "      <td>56650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295050</th>\n",
       "      <td>999891</td>\n",
       "      <td>TA208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295051</th>\n",
       "      <td>999892</td>\n",
       "      <td>P08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295052</th>\n",
       "      <td>999894</td>\n",
       "      <td>S95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295053</th>\n",
       "      <td>999931</td>\n",
       "      <td>39T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295054</th>\n",
       "      <td>999931</td>\n",
       "      <td>DSM 24617</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>295055 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id     strain\n",
       "0           100    DSM 101\n",
       "1        100053      56159\n",
       "2        100053      56640\n",
       "3        100053      56643\n",
       "4        100053      56650\n",
       "...         ...        ...\n",
       "295050   999891      TA208\n",
       "295051   999892        P08\n",
       "295052   999894        S95\n",
       "295053   999931        39T\n",
       "295054   999931  DSM 24617\n",
       "\n",
       "[295055 rows x 2 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Rename the index levels to 'name1' and 'name2'\n",
    "df_patric_unique = df_unique_assemblies_patric.rename_axis(index={'taxon_id': 'name1', 'strain': 'name2'}).reset_index()\n",
    "df_refseq_unique = df_unique_assemblies_refseq.rename_axis(index={'taxon_id': 'name1', 'strain': 'name2'}).reset_index()\n",
    "# Assuming the two DataFrames are df1 and df2\n",
    "common_taxon_strain = pd.merge(df_patric_unique[['taxon_id', 'strain']], df_refseq_unique[['taxon_id', 'strain']], \n",
    "                                on=['taxon_id', 'strain'], how='inner')\n",
    "\n",
    "# The result will contain the common unique combinations of 'taxon_id' and 'strain'\n",
    "common_taxon_strain\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>contigs_patric</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>contig_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>DSM 101</td>\n",
       "      <td>100.11</td>\n",
       "      <td>11</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100053</td>\n",
       "      <td>56159</td>\n",
       "      <td>100053.7</td>\n",
       "      <td>337</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100053</td>\n",
       "      <td>56640</td>\n",
       "      <td>100053.6</td>\n",
       "      <td>368</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100053</td>\n",
       "      <td>56643</td>\n",
       "      <td>100053.5</td>\n",
       "      <td>335</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100053</td>\n",
       "      <td>56650</td>\n",
       "      <td>100053.4</td>\n",
       "      <td>341</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295050</th>\n",
       "      <td>999891</td>\n",
       "      <td>TA208</td>\n",
       "      <td>999891.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295051</th>\n",
       "      <td>999892</td>\n",
       "      <td>P08</td>\n",
       "      <td>999892.3</td>\n",
       "      <td>22</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295052</th>\n",
       "      <td>999894</td>\n",
       "      <td>S95</td>\n",
       "      <td>999894.9</td>\n",
       "      <td>61</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295053</th>\n",
       "      <td>999931</td>\n",
       "      <td>39T</td>\n",
       "      <td>999931.5</td>\n",
       "      <td>2</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295054</th>\n",
       "      <td>999931</td>\n",
       "      <td>DSM 24617</td>\n",
       "      <td>999931.3</td>\n",
       "      <td>2</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>295055 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id     strain genome_id contigs_patric  \\\n",
       "0           100    DSM 101    100.11             11   \n",
       "1        100053      56159  100053.7            337   \n",
       "2        100053      56640  100053.6            368   \n",
       "3        100053      56643  100053.5            335   \n",
       "4        100053      56650  100053.4            341   \n",
       "...         ...        ...       ...            ...   \n",
       "295050   999891      TA208  999891.3              1   \n",
       "295051   999892        P08  999892.3             22   \n",
       "295052   999894        S95  999894.9             61   \n",
       "295053   999931        39T  999931.5              2   \n",
       "295054   999931  DSM 24617  999931.3              2   \n",
       "\n",
       "                                                 ftp_path contig_count  \n",
       "0       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           11  \n",
       "1       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          337  \n",
       "2       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          368  \n",
       "3       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          335  \n",
       "4       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          341  \n",
       "...                                                   ...          ...  \n",
       "295050  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "295051  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           22  \n",
       "295052  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           61  \n",
       "295053  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...            2  \n",
       "295054  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            2  \n",
       "\n",
       "[295055 rows x 6 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Perform the merge, keeping 'genome_id' from patric and 'ftp_path' from refseq\n",
    "common_taxon_strain = pd.merge(\n",
    "    df_patric_unique[['taxon_id', 'strain', 'genome_id','contigs']], \n",
    "    df_refseq_unique[['taxon_id', 'strain', 'ftp_path','contig_count']], \n",
    "    on=['taxon_id', 'strain'], \n",
    "    how='inner'\n",
    ")\n",
    "\n",
    "# The result will contain the common unique combinations of 'taxon_id' and 'strain'\n",
    "# along with 'genome_id' from patric and 'ftp_path' from refseq\n",
    "# Rename the 'contigs' column to 'contigs_patric'\n",
    "common_taxon_strain = common_taxon_strain.rename(columns={'contigs': 'contigs_patric'})\n",
    "\n",
    "# The result will have the 'contigs' column renamed to 'contigs_patric'\n",
    "common_taxon_strain\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>contigs_patric</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>contig_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>1001534</td>\n",
       "      <td>TI0902</td>\n",
       "      <td>1001534.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>1001542</td>\n",
       "      <td>TIGB03</td>\n",
       "      <td>1001542.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1001585</td>\n",
       "      <td>NK-01</td>\n",
       "      <td>1001585.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>100226</td>\n",
       "      <td>CFB_NBC_0001</td>\n",
       "      <td>100226.137</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>1002672</td>\n",
       "      <td>IMCC9063</td>\n",
       "      <td>1002672.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294973</th>\n",
       "      <td>99822</td>\n",
       "      <td>DB60705-15</td>\n",
       "      <td>99822.28</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294975</th>\n",
       "      <td>99822</td>\n",
       "      <td>NCTC4669</td>\n",
       "      <td>99822.25</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294989</th>\n",
       "      <td>99822</td>\n",
       "      <td>STREP97-15</td>\n",
       "      <td>99822.29</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294996</th>\n",
       "      <td>999183</td>\n",
       "      <td>DSM 25104</td>\n",
       "      <td>999183.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295050</th>\n",
       "      <td>999891</td>\n",
       "      <td>TA208</td>\n",
       "      <td>999891.3</td>\n",
       "      <td>1</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21109 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id        strain   genome_id contigs_patric  \\\n",
       "25      1001534        TI0902   1001534.3              1   \n",
       "26      1001542        TIGB03   1001542.3              1   \n",
       "29      1001585         NK-01   1001585.3              1   \n",
       "92       100226  CFB_NBC_0001  100226.137              1   \n",
       "103     1002672      IMCC9063   1002672.3              1   \n",
       "...         ...           ...         ...            ...   \n",
       "294973    99822    DB60705-15    99822.28              1   \n",
       "294975    99822      NCTC4669    99822.25              1   \n",
       "294989    99822    STREP97-15    99822.29              1   \n",
       "294996   999183     DSM 25104    999183.3              1   \n",
       "295050   999891         TA208    999891.3              1   \n",
       "\n",
       "                                                 ftp_path contig_count  \n",
       "25      https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "26      https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "29      https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "92      https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "103     https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "...                                                   ...          ...  \n",
       "294973  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "294975  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...            1  \n",
       "294989  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "294996  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "295050  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            1  \n",
       "\n",
       "[21109 rows x 6 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# First DataFrame where contigs_patric = \"1\" and contig_count = \"1\"\n",
    "df_contigs_one = common_taxon_strain[(common_taxon_strain['contigs_patric'] == \"1\") & \n",
    "                                     (common_taxon_strain['contig_count'] == \"1\")]\n",
    "\n",
    "# Second DataFrame containing the remaining rows\n",
    "df_remaining = common_taxon_strain[~((common_taxon_strain['contigs_patric'] == \"1\") & \n",
    "                                     (common_taxon_strain['contig_count'] == \"1\"))]\n",
    "\n",
    "# df_contigs_one contains rows where contigs_patric = \"1\" and contig_count = \"1\"\n",
    "# df_remaining contains the remaining rows\n",
    "df_contigs_one\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_remaining.to_csv(\"common_strains_with_multiple_contigs.csv\")\n",
    "df_contigs_one.to_csv(\"common_strains_with_one_contig.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>strain</th>\n",
       "      <th>genome_id</th>\n",
       "      <th>contigs_patric</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>contig_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100</td>\n",
       "      <td>DSM 101</td>\n",
       "      <td>100.11</td>\n",
       "      <td>11</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>100053</td>\n",
       "      <td>56159</td>\n",
       "      <td>100053.7</td>\n",
       "      <td>337</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100053</td>\n",
       "      <td>56640</td>\n",
       "      <td>100053.6</td>\n",
       "      <td>368</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100053</td>\n",
       "      <td>56643</td>\n",
       "      <td>100053.5</td>\n",
       "      <td>335</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>100053</td>\n",
       "      <td>56650</td>\n",
       "      <td>100053.4</td>\n",
       "      <td>341</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295049</th>\n",
       "      <td>999700</td>\n",
       "      <td>JA576,NBRC 108863,KCTC 15144</td>\n",
       "      <td>999700.3</td>\n",
       "      <td>69</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295051</th>\n",
       "      <td>999892</td>\n",
       "      <td>P08</td>\n",
       "      <td>999892.3</td>\n",
       "      <td>22</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295052</th>\n",
       "      <td>999894</td>\n",
       "      <td>S95</td>\n",
       "      <td>999894.9</td>\n",
       "      <td>61</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>61</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295053</th>\n",
       "      <td>999931</td>\n",
       "      <td>39T</td>\n",
       "      <td>999931.5</td>\n",
       "      <td>2</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295054</th>\n",
       "      <td>999931</td>\n",
       "      <td>DSM 24617</td>\n",
       "      <td>999931.3</td>\n",
       "      <td>2</td>\n",
       "      <td>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>273946 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       taxon_id                        strain genome_id contigs_patric  \\\n",
       "0           100                       DSM 101    100.11             11   \n",
       "1        100053                         56159  100053.7            337   \n",
       "2        100053                         56640  100053.6            368   \n",
       "3        100053                         56643  100053.5            335   \n",
       "4        100053                         56650  100053.4            341   \n",
       "...         ...                           ...       ...            ...   \n",
       "295049   999700  JA576,NBRC 108863,KCTC 15144  999700.3             69   \n",
       "295051   999892                           P08  999892.3             22   \n",
       "295052   999894                           S95  999894.9             61   \n",
       "295053   999931                           39T  999931.5              2   \n",
       "295054   999931                     DSM 24617  999931.3              2   \n",
       "\n",
       "                                                 ftp_path contig_count  \n",
       "0       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           11  \n",
       "1       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          337  \n",
       "2       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          368  \n",
       "3       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          335  \n",
       "4       https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...          341  \n",
       "...                                                   ...          ...  \n",
       "295049  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           69  \n",
       "295051  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           22  \n",
       "295052  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...           61  \n",
       "295053  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/9...            2  \n",
       "295054  https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/0...            2  \n",
       "\n",
       "[273946 rows x 6 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_remaining"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
